### PR TITLE
Add `scale` keyword argument to `mat_decompose`

### LINKED ISSUE #81: Add `scale` keyword argument to `mat_decompose`
In order to preserve the scaling component in a roundtrip from components -> matrix -> components, we need to be able to pass a `scale` vector into `mat_decompose`

```py
def mat_decompose(matrix, /, *, scale=None, dtype=None, out=None):
```

### PR DESCRIPTION
**Pull request by Korijn**
Closes #81 

- [x] Add test exemplifying problematic roundtrip
- [x] Add `scale` keyword argument to `mat_decompose`

### CODE REVIEW

**Comment by almarklein** [comment]
Would it be sufficient to only provide the signs of the scaling?

**Comment by Korijn** [comment]
> Would it be sufficient to only provide the signs of the scaling?

You are right. I've expanded the unit test to cover all possible enumerations of signs, and adjusted the function to only use the signs from the extra keyword argument.

File: pylinalg/matrix.py [hunk 1]
```diff
@@ -353,13 +358,23 @@ def mat_decompose(matrix, /, *, dtype=None, out=None):
...
 
+    if scaling_signs is not None:
+        # if the user provides the scaling signs, always use them
+        scaling_signs = np.sign(scaling_signs)
+    else:
+        # if not, detect if a flip is needed to reconstruct the transform
+        # and apply it to the first axis arbitrarily
+        flip = 1
+        if np.linalg.det(matrix) < 0:
+            flip = -1
```
Comment (almarklein) [inline pylinalg/matrix.py:369]: Just a bit shorter.
```suggestion
        flip = 1 if np.linalg.det(matrix) >= 0 else -1
```

**Comment by almarklein** [comment]
I wonder how we'd deal with cases where e.g. a pygfx both the object and parent have negative scaling signs. To create the world matrix, their matrices are multiplied. Should we separately multiply the `scale_signs`?

A related question, specific to the code here: should the function check whether the number of flipped axii matches the determinant? And what should it do if it does not match?

**Comment by Korijn** [comment]
> I wonder how we'd deal with cases where e.g. a pygfx both the object and parent have negative scaling signs. To create the world matrix, their matrices are multiplied. Should we separately multiply the `scale_signs`?

I think we're only calling compose and decompose to construct the local matrix using the local SRT components, not on the world matrix. So that shouldn't be an issue.

> A related question, specific to the code here: should the function check whether the number of flipped axii matches the determinant? And what should it do if it does not match?

I thought about it but I don't think that's something you can glean off the value of the determinant, actually.

**Comment by almarklein** [comment]
> I thought about it but I don't think that's something you can glean off the value of the determinant, actually.

Would it not be something like, if the det is smaller than zero, that the number of negative axii should be uneven?

**Comment by Korijn** [comment]
I've added the validation as far as I think we can provide it, and an accompanying unit test.

**Review by almarklein — APPROVED** [review]
Cool, let's see how this works out in pygfx!

**Pull request merged**